src/Spago/Build.hs

```diff
@@ -84,13 +86,33 @@ build BuildOptions{..} maybePostBuild = do
     Watch     -> Watch.watch (Set.fromAscList $ fmap Glob.compile absoluteProjectGlobs) shouldClear buildAction
 
 -- | Start a repl
-repl :: Spago m => [Purs.SourcePath] -> [Purs.ExtraArg] -> m ()
-repl sourcePaths passthroughArgs = do
+repl :: Spago m => Maybe Int -> Maybe GlobalCache.CacheFlag -> [PackageSet.PackageName] -> [Purs.SourcePath] -> [Purs.ExtraArg] -> m ()
+repl jobsLimit cacheFlag newPackages sourcePaths passthroughArgs = do
   echoDebug "Running `spago repl`"
-  config <- Config.ensureConfig
-  deps <- Packages.getProjectDeps config
-  let globs = Packages.getGlobs deps <> Config.configSourcePaths config <> sourcePaths
-  Purs.repl globs passthroughArgs
+
+  try Config.ensureConfig >>= \case
+    Right config -> do
+      deps <- Packages.getProjectDeps config
+      let globs = Packages.getGlobs deps <> Config.configSourcePaths config <> sourcePaths
+      Purs.repl globs passthroughArgs
+    Left (err :: SpagoError) -> do
```

Comment:
I'm new to exceptions in Haskell. Is it correct to catch `SpagoError` here?

Reply:
Good question. In Haskell the catchall exception is `SomeException`, and we usually just catch that around the project (and I'd also catch that here)

The reason why you don't want to catch it everywhere is that sometimes you're better off just failing on Async exceptions (which are caught by `SomeException` too). I find async exceptions to be a very tricky part of Haskell to get right. You can read more about them [here](https://simonmar.github.io/posts/2017-01-24-asynchronous-exceptions.html) or [here](https://www.fpcomplete.com/blog/2018/04/async-exception-handling-haskell)

Reply:
I see, thanks for the pointers!

Reply:
You're welcome 🙂